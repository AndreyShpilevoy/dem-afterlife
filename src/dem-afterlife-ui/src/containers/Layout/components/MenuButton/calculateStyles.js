const calculateStyles = ({ menuButton, focus }) => ({
  container: {
    cursor: menuButton.cursor,
    height: menuButton.widthAndHeight,
    transform: "rotate(0deg)",
    width: menuButton.widthAndHeight,
    "&:focus": {
      outlineColor: focus.colorLight
    }
  },
  allLines: {
    borderRadius: "3.333%",
    background: menuButton.line.color,
    height: "10%",
    left: 0,
    marginLeft: "10%",
    position: "absolute",
    transform: "rotate(0deg)",
    transition: menuButton.line.transition,
    width: "80%"
  },
  firstLine: {
    top: "20%",
    "&.open": {
      top: "45%",
      width: "0%",
      left: "50%"
    }
  },
  secondLine: {
    top: "45%",
    "&.open": {
      transform: "rotate(45deg)"
    }
  },
  thirdLine: {
    top: "45%",
    "&.open": {
      transform: "rotate(-45deg)"
    }
  },
  fourthLine: {
    top: "70%",
    "&.open": {
      top: "45%",
      width: "0%",
      left: "50%"
    }
  }
});

export default calculateStyles;
