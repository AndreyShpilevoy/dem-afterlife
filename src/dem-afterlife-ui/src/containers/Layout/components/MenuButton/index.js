/* eslint fp/no-class: 0, fp/no-nil: 0, fp/no-unused-expression: 0, fp/no-this: 0 */

import React, { Component } from "react";
import { func, shape } from "prop-types";
import { concatStyleNames, injectSheet } from "styles";
import { handleEnterKeyDown } from "utils";
import calculateStyles from "./calculateStyles";

export class MenuButtonPure extends Component {
  static propTypes = {
    classes: shape().isRequired,
    onClick: func.isRequired
  };

  constructor(props) {
    super(props);
    this.state = { opened: false };
  }

  menuButtonOnClick = () => {
    const { opened } = this.state;
    const { onClick } = this.props;
    this.setState({ opened: !opened });
    if (onClick) {
      onClick();
    }
  };

  handleKeyDown = event =>
    handleEnterKeyDown(event) ? this.menuButtonOnClick() : () => ({});

  render() {
    const { classes } = this.props;
    const {
      container,
      firstLine,
      secondLine,
      thirdLine,
      fourthLine,
      allLines
    } = classes;
    const { opened } = this.state;
    const openedClassName = opened ? "open" : "";
    const firstLineStyle = concatStyleNames([
      firstLine,
      allLines,
      openedClassName
    ]);
    const secondLineStyle = concatStyleNames([
      secondLine,
      allLines,
      openedClassName
    ]);
    const thirdLineStyle = concatStyleNames([
      thirdLine,
      allLines,
      openedClassName
    ]);
    const fourthLineStyle = concatStyleNames([
      fourthLine,
      allLines,
      openedClassName
    ]);
    return (
      <div
        className={container}
        onClick={this.menuButtonOnClick}
        onKeyDown={this.handleKeyDown}
        role="button"
        tabIndex={0}
      >
        <div className={firstLineStyle} />
        <div className={secondLineStyle} />
        <div className={thirdLineStyle} />
        <div className={fourthLineStyle} />
      </div>
    );
  }
}

export default injectSheet(calculateStyles, { componentName: "MenuButton" })(
  MenuButtonPure
);
