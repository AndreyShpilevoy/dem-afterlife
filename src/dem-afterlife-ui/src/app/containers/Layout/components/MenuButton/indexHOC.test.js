/* eslint no-undef: 0, fp/no-unused-expression: 0, fp/no-nil: 0, fp/no-mutation:0*/

import React from 'react';
import {shallow} from 'enzyme';
import * as ThemedStyleSheet from 'react-with-styles/lib/ThemedStyleSheet';
import MenuButton from './index';

ThemedStyleSheet.default.get = () => {};

describe('MenuButton HOC', () => {
    it('match expected snapshot', () => {
        expect(shallow(<MenuButton theme={'default'} />)).toMatchSnapshot();
    });
});
