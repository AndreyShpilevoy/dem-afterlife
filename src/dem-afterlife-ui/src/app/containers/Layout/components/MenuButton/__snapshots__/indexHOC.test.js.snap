// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuButton HOC match expected snapshot 1`] = `
<MenuButtonPure
  onClick={[Function]}
  styles={
    Object {
      ".default-MenuButton-container:focus": "default-MenuButton-container:focus",
      ".default-MenuButton-firstLine.open": "default-MenuButton-firstLine.open",
      ".default-MenuButton-fourthLine.open": "default-MenuButton-fourthLine.open",
      ".default-MenuButton-secondLine.open": "default-MenuButton-secondLine.open",
      ".default-MenuButton-thirdLine.open": "default-MenuButton-thirdLine.open",
      "allLines": "default-MenuButton-allLines",
      "container": "default-MenuButton-container",
      "firstLine": "default-MenuButton-firstLine",
      "fourthLine": "default-MenuButton-fourthLine",
      "secondLine": "default-MenuButton-secondLine",
      "thirdLine": "default-MenuButton-thirdLine",
    }
  }
/>
`;
